neutron/convert.cpp

```diff
@@ -245,16 +240,57 @@ template <class T> T convert_impl(T d, const Dim from, const Dim to) {
       "Conversion between requested dimensions not implemented yet.");
 }
 
+namespace {
+template <class T>
+T swap_tof_related_labels_and_attrs(T &&x, const Dim from, const Dim to) {
+  auto fields = {"position", "source_position", "sample_position"};
+  // TODO Add `extract` methods to do this in one step and avoid copies?
+  if (from == Dim::Tof) {
+    for (const auto &field : fields) {
+      if (x.labels().contains(field)) {
+        // TODO This is an unfortunate duplication of attributes. It is
+        // (currently?) required due to a limitation of handling attributes of
+        // Dataset and its items *independently* (no mapping of dataset
+        // attributes into item attributes occurs, unlike for coords and
+        // labels). If we did not also add the attributes to each of the items,
+        // a subsequent unit conversion of an item on its own would not be
+        // possible. It needs to be determined if there is a better way to
+        // handle attributes so this can be avoided.
```

Comment:
Not for this PR, but we have two cases at present. Either the variables entirely match (labels), or they have not match restrictions (attributes). For the problem here, do we need to consider the possibility of allowing partial matches? Different datasets are allowed to contain different (or overlapping) portions of the instrument, but for any overlap the Dim.Position values must be identical.

Reply:
This is a discussion related to http://xarray.pydata.org/en/stable/computation.html#automatic-alignment which scipp is not supporting right now.